File: CPP/3208. Alternating Groups II/solution.cpp
class Solution {
    public:
        int numberOfAlternatingGroups(vector<int>& colors, int k) {
            int n = colors.size();
            int count = 0;
    
            int altCount = 1;
            for (int i = 1; i < k; i++) {
                if (colors[i] != colors[i - 1]) altCount++;
            }
            if (altCount == k) count++;
    
            for (int i = 1; i < n; i++) {
                if (colors[i] != colors[i - 1]) altCount--;
                if (colors[(i + k - 1) % n] != colors[(i + k - 2) % n]) altCount++;
                
                if (altCount == k) count++;
            }
    
            return count;
        }
    };

File: Python/3208. Alternating Groups II/solution.py
class Solution(object):
    def numberOfAlternatingGroups(self, colors, k):
        n = len(colors)
        if k > n:
            return 0
        extended_colors = colors + colors[:k-1]
        count, length = 0, 1

        for i in range(1, len(extended_colors)):
            if extended_colors[i] != extended_colors[i - 1]:
                length += 1
            else:
                length = 1

            if length >= k:
                count += 1

        return count